cpp/oneapi/dal/algo/basic_statistics/compute_types.hpp

```diff
@@ -0,0 +1,211 @@
+template <typename Task = task::by_default>
+class compute_result : public base {
+    static_assert(detail::is_valid_task_v<Task>);
+
+public:
+    using task_t = Task;
+
+    /// Creates a new instance of the class with the default property values.
+    compute_result();
+
+    /// A $n1 \\times n2$ table with the minimum result.
+    /// @remark default = table{}
+    const table& get_min() const;
+
+    /// A $n1 \\times n2$ table with the maximum result.
+    /// @remark default = table{}
+    const table& get_max() const;
+
+    /// A $n1 \\times n2$ table with the sum result.
+    /// @remark default = table{}
+    const table& get_sum() const;
+
+    /// A $n1 \\times n2$ table with the sum_squares result.
+    /// @remark default = table{}
+    const table& get_sum_squares() const;
+
+    /// A $n1 \\times n2$ table with the sum_squares_centered result.
+    /// @remark default = table{}
+    const table& get_sum_squares_centered() const;
+
+    /// A $n1 \\times n2$ table with the mean result.
+    /// @remark default = table{}
+    const table& get_mean() const;
+
+    /// A $n1 \\times n2$ table with the second_order_raw_moment result.
+    /// @remark default = table{}
+    const table& get_second_order_raw_moment() const;
+
+    /// A $n1 \\times n2$ table with the variance result.
+    /// @remark default = table{}
+    const table& get_variance() const;
+
+    /// A $n1 \\times n2$ table with the standard_deviation result.
+    /// @remark default = table{}
+    const table& get_standard_deviation() const;
+
+    /// A $n1 \\times n2$ table with the variation result.
+    /// @remark default = table{}
+    const table& get_variation() const;
+
+    auto& set_min(const table& value) {
+        set_min_impl(value);
+        return *this;
+    }
+
+    auto& set_max(const table& value) {
+        set_max_impl(value);
+        return *this;
+    }
+
+    auto& set_sum(const table& value) {
+        set_sum_impl(value);
+        return *this;
+    }
+
+    auto& set_sum_squares(const table& value) {
+        set_sum_squares_impl(value);
+        return *this;
+    }
+
+    auto& set_sum_squares_centered(const table& value) {
+        set_sum_squares_centered_impl(value);
+        return *this;
+    }
+
+    auto& set_mean(const table& value) {
+        set_mean_impl(value);
+        return *this;
+    }
+
+    auto& set_second_order_raw_moment(const table& value) {
+        set_second_order_raw_moment_impl(value);
+        return *this;
+    }
+
+    auto& set_variance(const table& value) {
+        set_variance_impl(value);
+        return *this;
+    }
+
+    auto& set_standard_deviation(const table& value) {
+        set_standard_deviation_impl(value);
+        return *this;
+    }
+
+    auto& set_variation(const table& value) {
+        set_variation_impl(value);
+        return *this;
+    }
+
+    /// Result options that indicates availability of the properties
+    /// @remark default = default_result_options<Task>
```

Comment:
Probably, this is error in the source from which the comment was copied. There is no `default_result_options<Task>` in the public API. Need to be fixed as soon as possible to not let it spread out across all code base.

Reply:
fixed here, and will update in KNN.